File: android/entities/src/main/java/com/tiagohs/entities/enums/ContentType.kt
package com.tiagohs.entities.enums

import com.google.gson.annotations.SerializedName
import java.io.Serializable

enum class ContentType(
    val type: String
): Serializable {
    @SerializedName("text")
    TEXT("text"),

    @SerializedName("gif")
    GIF("gif"),

    @SerializedName("video")
    VIDEO("video"),

    @SerializedName("slide")
    SLIDE("slide"),

    @SerializedName("image")
    IMAGE("image"),

    @SerializedName("audio_stream")
    AUDIO_STREAM("audio_stream"),

    @SerializedName("quote")
    QUOTE("quote"),

    @SerializedName("block_special")
    BLOCK_SPECIAL("block_special"),

    @SerializedName("link_screen")
    LINK_SCREEN("link_screen"),

    @SerializedName("movie_list")
    MOVIE_LIST("movie_list"),

    @SerializedName("person_list")
    PERSON_LIST("person_list"),

    @SerializedName("recomendations")
    RECOMENDATIONS("recomendations"),

    @SerializedName("awards_nominees")
    AWARDS_NOMINEES("awards_nominees"),

    @SerializedName("movie_list_special")
    MOVIE_LIST_SPECIAL("movie_list_special");

    companion object {
        fun getContentType(type: String): ContentType {
            var typeEnum = TEXT

            for (typeValue in values()) {
                if (typeValue.type.equals(type)) {
                    typeEnum = typeValue
                    break
                }
            }

            return typeEnum
        }
    }
}

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/Enums/ContentType.swift
//
//  ContentType.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 21/02/21.
//

import Foundation

enum ContentType: String {
    case text
    case gif
    case video
    case slide
    case image
    case audio_strem
    case quote
    case block_special
    case link_screen
    case movie_list
    case person_list
    case recomendations
    case awards_nominees
    case movie_list_special
    
    static func getContentType(by name: String) -> ContentType {
        switch name {
        case "text":
            return .text
        case "gif":
            return .gif
        case "video":
            return .video
        case "slide":
            return .slide
        case "image":
            return .image
        case "audio_strem":
            return .audio_strem
        case "quote":
            return .quote
        case "block_special":
            return .block_special
        case "link_screen":
            return .link_screen
        case "movie_list":
            return .movie_list
        case "person_list":
            return .person_list
        case "recomendations":
            return .recomendations
        case "awards_nominees":
            return .awards_nominees
        case "movie_list_special":
            return .movie_list_special
        default:
            return .text
        }
    }
}
